client/src/modules/Registry/pages/Student/index.tsx

```diff
@@ -26,24 +27,40 @@ export type Props = {
   session: Session;
 };
 
+interface IdName {
+  id: number;
+  name: string;
+}
+
 export function StudentRegistry(props: Props & { courseAlias?: string }) {
+  const { githubId } = props.session;
   const [form] = Form.useForm();
   const update = useUpdate();
   const [loading, setLoading] = useState(false);
   const [submitted, setSubmitted] = useState(false);
   const [courses, setCourses] = useState([] as Course[]);
+  const [registeredForCourses, setRegisteredForCourses] = useState([] as IdName[]);
   const [location, setLocation] = useState(null as Location | null);
   const [initialData, setInitialData] = useState({} as Partial<UserFull>);
 
   useAsync(async () => {
     setLoading(true);
     const userService = new UserService();
     const courseService = new CoursesService();
-    const [profile, courses] = await Promise.all([userService.getMyProfile(), courseService.getCourses()]);
+    const [profile, profileInfo, courses] = await Promise.all([
+      userService.getMyProfile(),
+      userService.getProfileInfo(githubId),
```

Comment:
why do you need to pull profile info data? Everything should be available in `props.session`. Isn't it?

Reply:
Can we get the values for the course isExpelled and isCourseCompleted from session? I have not found a way to do this.